dist/tools/prep/prep.py

```diff
@@ -0,0 +1,214 @@
+#!/usr/bin/env python3
+
+# Copyright (C) 2015  Hauke Petersen <dev@haukepetersen.de>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import sys
+import os
+from os import path
+import argparse
+import yaml
+import templates
+
+# Configuration data
+conf_file = path.expanduser("~") + "/.riot/prep_conf.yaml"
+
+# Value definition for configuration
+conf_items = {
+    'name': 'Your full name',
+    'email': 'Your email address',
+    'copyright': 'Your copyright',
+    'riotbase': 'Absolute path to the RIOT base directory'
+}
+
+driver_items = {
+    'id': 'Device name, used as module name in RIOT',
+    'name': 'Human readable device descriptor, e.g. "ISL2920 light sensor"'
+}
+
+sys_items = {
+    'name': 'Name of the new module',
+    'desc': 'Short description of the module'
+}
+
+def load_config( ):
+    if not path.isfile(conf_file):
+        return {}
+    else:
+        with open(conf_file) as f:
+            return yaml.load(f)
+
+def store_config( conf ):
+    tmp = yaml.dump(conf, default_flow_style=False)
+    with open(conf_file, "w") as f:
+        f.write(tmp)
+
+
+def askfor( var, items ):
+    print("Now just fill in the required information. Just press return to")
+    print("use the default value that is given in parenthesis\n")
+    for item, desc in items.items():
+        if item in var.keys():
+            val = var[item]
+        else:
+            val = ''
+        print(desc + " (" + val + "):")
+        line = sys.stdin.readline().rstrip()
+        if line:
+            var[item] = line
+        else:
+            var[item] = val
+
+def make_makefile( location ):
+    name = "%s/Makefile" % (location)
+    if path.isfile(name):
+        print("error: file '%s' does already exist" % (name))
+        return
+    with open(name, "w") as f:
+        f.write("include $(RIOTBASE)/Makefile.base\n")
+
+def make_mainheader( filename, copyright, group, module, desc, name, email ):
```

Comment:
You should use string.Template here. That way, you can supply a dictionary, e.g.,

``` {'filename'  : 'xxx', 'copyright:' : 'yyy', ...```

... and just use ```${filename}``` within the template file.

That way, you can write one generic template file funtion, and then use that in e.g., make_makefile, just checking if the dictionary has the correct entries, and thenn hand over to the generic function.